administrative-sdk/pronunciationChallenge.js

```diff
@@ -115,150 +77,86 @@ class PronunciationChallenge {
-
-  /**
-   * Error callback used by listPronunciationChallenges.
-   *
-   * @callback Sdk~listPronunciationChallengesErrorCallback
-   * @param {object[]} errors Array of errors.
-   */
-  listPronunciationChallengesErrorCallback(errors) {}
-
-  /**
    * List all pronunciation challenges in the organisation.
    *
    * @param {string} organisationId Specify an organisation identifier.
    * @param {Sdk~listPronunciationChallengesCallback} cb The callback that handles the response.
    * @param {Sdk~listPronunciationChallengesErrorCallback} [ecb] The callback that handles the error response.
    */
-  static listPronunciationChallenges(
-    connection, organisationId, cb, ecb) {
-    var _cb = function(data) {
-      var challenges = [];
-      data.forEach(function(datum) {
-        var challenge = new PronunciationChallenge(
-          organisationId, datum.id, datum.transcription);
-        challenge.created = new Date(datum.created);
-        challenge.updated = new Date(datum.updated);
-        challenge.referenceAudioUrl = datum.referenceAudioUrl;
-        challenge.status = datum.status;
-        challenges.push(challenge);
-      });
-      if (cb) {
-        cb(challenges);
-      }
-    };
-
+  static listPronunciationChallenges(connection, organisationId) {
     var url = connection.settings.apiUrl + '/organisations/' +
       organisationId + '/challenges/pronunciation';
-    connection._secureAjaxGet(url, _cb, ecb);
+    return connection._secureAjaxGet(url)
+      .then(data => {
+        var challenges = [];
+        data.forEach(function(datum) {
+          var challenge = new PronunciationChallenge(
+            organisationId, datum.id, datum.transcription);
+          challenge.created = new Date(datum.created);
+          challenge.updated = new Date(datum.updated);
+          challenge.referenceAudioUrl = datum.referenceAudioUrl;
+          challenge.status = datum.status;
+          challenges.push(challenge);
+        });
+        return challenges;
+      });
   }
 
   /**
-   * Callback used by deletePronunciationChallenge.
-   *
-   * @callback Sdk~pronunciationChallengeDeletedCallback
-   */
-  pronunciationChallengeDeletedCallback(challenge) {}
-
-  /**
-   * Error callback used by deletePronunciationChallenge.
-   *
-   * @callback Sdk~pronunciationChallengeDeletedErrorCallback
-   * @param {object[]} errors Array of errors.
-   * @param {its.PronunciationChallenge} challenge Pronunciation challenge domain model instance with unapplied changes.
-   */
-  pronunciationChallengeDeletedErrorCallback(errors, challenge) {}
-
-  /**
    * Delete a pronunciation challenge.
    *
    * @param {its.PronunciationChallenge} challenge A pronunciation challenge object.
    * @param {Sdk~pronunciationChallengeDeletedCallback} [cb] The callback that handles the response.
    * @param {Sdk~pronunciationChallengeDeletedErrorCallback} [ecb] The callback that handles the error response.
```

Comment:
Missing documentation update. Callbacks are no longer taken and a promise is returned.